### ISSUE TITLE
Confusion matrix with class names

### ISSUE THREAD

**Issue by Juanlu001** [rasbt/mlxtend#485]
I came here looking for a quick way to plot a confusion matrix with class labels, but found that there's not such a way. I think it would be a great addition, at least to the documentation. For the moment, I'll be using scikit-plot.

**Comment by rasbt** [comment]
Good point, I can add it to the documentation. I could add it as function parameters, but it's maybe overkill. The modification would basically be 

```python
fig, ax = plot_confusion_matrix(conf_matrix)
ax.set_xticklabels(['', 'Positive Class (0)', 'Negative Class (1)'])
ax.set_yticklabels(['', 'Positive Class (0)', 'Negative Class (1)']);
```

**Comment by rasbt** [comment]
I see. You mean if you don't have integer class labels like

```python
y_target =    [1, 1, 1, 0, 0, 2, 0, 3]
y_predicted = [1, 0, 1, 0, 0, 2, 1, 3]
```
but some string variables in unknown order? In that case it will be ordered alphabetically.


import matplotlib.pyplot as plt
from mlxtend.evaluate import confusion_matrix
import numpy as np

```python
y_target =    ['b', 'b', 'b', 'a', 'a', 'c', 'a', 'd']
y_predicted = ['b', 'a', 'a', 'a', 'a', 'c', 'a', 'd']

cm = confusion_matrix(y_target=y_target, 
                      y_predicted=y_predicted,)
cm
```

```array([[3, 0, 0, 0],
       [2, 1, 0, 0],
       [0, 0, 1, 0],
       [0, 0, 0, 1]])
```

and to plot it:

```python
from mlxtend.plotting import plot_confusion_matrix

fig, ax = plot_confusion_matrix(cm)

ax.set_xticklabels([''] + list(np.unique(y_target)))
ax.set_yticklabels([''] + list(np.unique(y_target)))
plt.show()
```

![unknown](https://user-images.githubusercontent.com/5618407/50352822-026daf80-050c-11e9-92ff-f211e3884e54.png)

**Comment by rasbt** [comment]
Has been addressed now via #487

**Comment by Juanlu001** [comment]
Awesome, thanks!
